activerecord/test/cases/adapters/sqlite3/quoting_test.rb

```diff
@@ -15,11 +15,19 @@ def test_type_cast_binary_encoding_without_logger
   end
 
   def test_type_cast_true
+    ActiveRecord::Base.represent_sqlite3_boolean_as_integer = false
     assert_equal "t", @conn.type_cast(true)
+
+    ActiveRecord::Base.represent_sqlite3_boolean_as_integer = true
+    assert_equal 1, @conn.type_cast(true)
   end
 
   def test_type_cast_false
+    ActiveRecord::Base.represent_sqlite3_boolean_as_integer = false
     assert_equal "f", @conn.type_cast(false)
+
+    ActiveRecord::Base.represent_sqlite3_boolean_as_integer = true
+    assert_equal 0, @conn.type_cast(false)
```

Comment:
These tests should make sure to leave the flag in the state they found it.. while it's likely nothing else cares now, it could cause an accidental test-ordering dependency later.